<?xml version="1.0"?>
<launch>

  <arg name="input_clouds_not_mesh" default="true" />

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find gaitmesh_ros)/rviz/annotate_cloud_gaits.rviz" />

  <node pkg="gaitmesh_ros" type="annotate_cloud_gaits" name="annotate_cloud_gaits" output="screen">
    <param name="input_clouds_not_mesh" value="$(arg input_clouds_not_mesh)" />

    <param name="radius_normals" value="0.20" />
    <param name="radius_robot" value="0.40" />
    <param name="curvature_threshold" value="0.04" />
    <param name="mesh_sampling_resolution" value="0.10" />

    <!-- example model from Barcelona Robot Lab Dataset -->
<!--
    <param name="mesh" value="$(find gaitmesh_ros)/data/BRL_final.obj" />
-->
    <!-- oil_rig model -->
<!--
    <param name="mesh" value="$(find recast_demos)/data/oilRig_detailed.obj" />
-->
    <!-- Blyth #ORCAHubDemos -->

    <param name="cloud1" value="/home/martim/ORCAHubDemos/blyth-01-10-2019-5cm-downsampled-cleaned2-cutceiling-cutside-nopillars-cleanfloor.ply" />
    <param name="cloud_mesh" value="/home/martim/ORCAHubDemos/mesh-blyth-01-10-2019-5cm-downsampled-cleaned2-cutceiling-cutside-nopillars-cleanfloor.obj" />

    <!-- FSC full model -->
<!--
    <param name="cloud1" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-1stfloor-1cm-clipped-icp.ply" />
    <param name="cloud2" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-merged-main-1cm-clipped-icp-clean.ply" />
    <param name="cloud3" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-west-1cm-clipped-icp.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-merged-all.obj" />
-->
    <!-- FSC main floor only -->
<!--
    <param name="cloud1" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-main-1cm-clipped-icp-clean3.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-merged-main-1cm-clipped-icp-clean3.obj" />
-->
    <!-- FSC container1 only -->
<!--
    <param name="cloud2" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-container-merged-1cm-aligned-normals-ascii-filtered.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-container-merged-1cm-aligned-normals-ascii-filtered.obj" />
-->
    <!-- FSC container1 and main -->
<!--
    <param name="cloud1" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-main-1cm-clipped-icp-clean3.ply" />
    <param name="cloud2" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-container-merged-1cm-aligned-normals-ascii-filtered-realign-cutmain.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-container-main-merged.obj" />
-->
    <!-- FSC container1 and full model -->
<!--
    <param name="cloud1" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-1stfloor-1cm-clipped-icp.ply" />
    <param name="cloud2" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-main-1cm-clipped-icp-clean3.ply" />
    <param name="cloud3" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-merged-west-1cm-clipped-icp-clean.ply" />
    <param name="cloud4" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/oilrig-container-merged-1cm-aligned-normals-ascii-filtered-realign-cutmain.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/fire-service-college/extended_map/mesh-oilrig-final-main-container-1stfloor-west.obj" />
-->
    <!-- ORI map -->
<!--
    <param name="cloud1" value="/home/martim/drs_testing_data/leica/ori-building-maps/map_filter_cutaway_for_autonomy_5cm.ply" />
    <param name="cloud_mesh" value="/home/martim/drs_testing_data/leica/ori-building-maps/mesh_holefilled_map_filter_cutaway_for_autonomy_5cm.obj" />
-->
  </node>

</launch>
